### ISSUE TITLE
[Bug]: Open ann.db file created with python annoy

### ISSUE THREAD

**Issue by mau-io** [prathameshnetake/annoy-node#2]
SO: Windows 10

First i create annoy file with python:

```python
from annoy import AnnoyIndex
import random

f = 40
t = AnnoyIndex(f, 'angular')  # Length of item vector that will be indexed

for i in range(1000):
    v = [random.gauss(0, 1) for z in range(f)]
    t.add_item(i, v)

t.build(100) # 100 trees
t.save('C:/Users/github/Documents/annoy/test.ann')

u = AnnoyIndex(f, 'angular')
u.load('C:/Users/github/Documents/annoy/test.ann') # super fast, will just mmap the file
print(u.get_nns_by_item(0, 10)) # will find the 1000 nearest neighbors
```

But if I want to load the file using annoy-node I always get the same error:
```javascript
const Annoy = require("annoy-node");
const annoy = new Annoy.default(40, Annoy.Metric.ANGULAR);

annoy.load('C:/Users/github/Documents/annoy/test.ann');

const result = annoy.get_nns_by_item(2, 1, true);
console.log(result);
```

**_Index size is not a multiple of vector size. Ensure you are opening using the same metric you used to create the index.: No error (0)
Segmentation fault_**

Any tips about how to get things going would be greatly appreciated!

**Comment by prathameshnetake** [comment]
Did you try to open this file by CPP?

**Comment by LukaszSolo** [comment]
I stumbled upon this too, and so far what I found is that Python package casts vectors as floats (byte size 4),
https://github.com/spotify/annoy/blob/6f6b0c84ab413337eb4d2e850a4cba637f52ccbc/annoy/__init__.pyi#L6

Same with another node module, it stores data as float:
https://github.com/jimkang/annoy-node/blob/3e9ecb91f3e6029b37a2d362397f8654f5c27120/annoyindexwrapper.cc#L20

While this wrappers saves them as doubles (byte size 8)

https://github.com/prathameshnetake/annoy-node/blob/70be7181d334908d2a91f559d4ba5d64952b9526/src/annoy.cpp#L63

Since C++ backend is templated, it accepts any type as long as they are in sync. That's why python and other node works together, while not with `annoy-node`

@prathameshnetake Is there a reason you decided to go with double?

@mau-io Solution (at least to be able to load) is to replace Float64Array with Float32Array, plus change corresponding places to be `float` instead of `double`

**Comment by prathameshnetake** [comment]
@LukaszSolo Nice catch with double type

**Comment by prathameshnetake** [comment]
Issue is fixed on 0.2.0
